apps/test-viewer/src/components/Viewer.tsx

```diff
@@ -0,0 +1,123 @@
+  const { iTwinId, iModelId } = useIModelInfo();
+
+  return (
+    <WebViewer
+      iTwinId={iTwinId ?? ""}
+      iModelId={iModelId ?? ""}
+      authClient={authClient}
+      viewCreatorOptions={viewCreatorOptions}
+      enablePerformanceMonitors={false}
+      onIModelAppInit={onIModelAppInit}
+      uiProviders={[
+        ...uiConfig.uiItemsProviders
+      ]}
+      defaultUiConfig={{
+        hideNavigationAid: true,
+        hideStatusBar: true,
+        hideToolSettings: true
+      }}
+    />
+  );
+}
+
+function useViewCreatorOptions() {
+  /** NOTE: This function will execute the "Fit View" tool after the iModel is loaded into the Viewer.
+   * This will provide an "optimal" view of the model. However, it will override any default views that are
+   * stored in the iModel. Delete this function and the prop that it is passed to if you prefer
+   * to honor default views when they are present instead (the Viewer will still apply a similar function to iModels that do not have a default view).
+   */
+  const viewConfiguration = useCallback((viewPort: ScreenViewport) => {
+    // default execute the fitview tool and use the iso standard view after tile trees are loaded
+    const tileTreesLoaded = () => {
+      return new Promise((resolve, reject) => {
+        const start = new Date();
+        const intvl = setInterval(() => {
+          if (viewPort.areAllTileTreesLoaded) {
+            ViewerPerformance.addMark("TilesLoaded");
+            ViewerPerformance.addMeasure(
+              "TileTreesLoaded",
+              "ViewerStarting",
+              "TilesLoaded"
+            );
+            clearInterval(intvl);
+            resolve(true);
+          }
+          const now = new Date();
+          // after 20 seconds, stop waiting and fit the view
+          if (now.getTime() - start.getTime() > 20000) {
+            reject();
+          }
+        }, 100);
+      });
+    };
+
+    tileTreesLoaded().finally(() => {
+      void IModelApp.tools.run(FitViewTool.toolId, viewPort, true, false);
+      viewPort.view.setStandardRotation(StandardViewId.Iso);
+    });
+  }, []);
+
+  return useMemo(
+    () => ({ viewportConfigurer: viewConfiguration }),
+    [viewConfiguration]
+  );
+}
+
+function useIModelInfo() {
+  const [iModelId, setIModelId] = useState(process.env.IMJS_IMODEL_ID);
+  const [iTwinId, setITwinId] = useState(process.env.IMJS_ITWIN_ID);
+
+  useEffect(() => {
+    const urlParams = new URLSearchParams(window.location.search);
+    if (urlParams.has("iTwinId")) {
+      setITwinId(urlParams.get("iTwinId") as string);
+    } else {
+      if (!process.env.IMJS_ITWIN_ID) {
+        throw new Error(
+          "Please add a valid iTwin ID in the .env file and restart the application or add it to the iTwinId query parameter in the url and refresh the page. See the README for more information."
+        );
+      }
+    }
+
+    if (urlParams.has("iModelId")) {
+      setIModelId(urlParams.get("iModelId") as string);
```

Comment:
dont need to check/throw error if imodel id is missing in .env?

Reply:
This code is from cra-web-viewer-template I only moved it around to make component smaller. I can add check if you think it's necessary.

Reply:
It's fine. Resolving...
I'll make sure we update the cra template